### DISCUSSION TITLE
Zwift/Companion outdoor rides

### DISCUSSION THREAD

**Discussion by eskimos20** [netbrain/zwift#159] [Q&A]
Hi Everyone! 

I got this Idea that I wanted to:

* Start zwift in docker at a PC at home.. say (192.168.0.2, LAN)
* Have a VPN server either on the same server (192.168.0.2, LAN) or another machine say (192.168.0.3, LAN)
* Go outside, connect the phone to that VPN-server that has access to all IPs on the LAN
* Start Companion app and connect my powermeter/heartrate monitor via the companion app to Zwift running at home.

The trouble I have noticed is that I think the companion app/zwift is broadcasting over multicast or something(not sure). But they cant find each other when connected to VPN, but can find each other when connected to wireless/or wired network at home without VPN.

Am I totally off here or is it a possibility to get it to work ? 

I am currently using wireguard.

/Eskimos20

**Comment by netbrain** [comment]
Yeah that should work i think, not sure how you would set it up with wireguard + vpn, but I think it would be doable with tailscale, no VPN required.

Good luck! have fun :)

**Reply by eskimos20** [reply]
Thanks! 

I tried Tailscale.. very easy setup. But it seems that multicast is not supported yet. 

Hope to find a solution in the future :)

**Reply by netbrain** [reply]
Maybe this would be relevant?
https://github.com/troglobit/smcroute

Apparently can create routing configuration for multicast.

I'm guessing what you want is the possibility to take a source multicast packet, then route it over a specific VPN solution to a IP, and then let that handle the TCP packet and translate it to multicast again? I.E
```

              +--------------------------+
              |    Multicast Source      |
              |     (224.0.0.1)          |
              +--------------------------+
                       |
                       | multicast packets
                       |
              +--------------------------+
              |     Conversion Device    |
              | (Multicast to TCP/IP)    |
              +--------------------------+
                       |
                       | TCP/IP over VPN
                       |
              +--------------------------+
              |       VPN Tunnel         |
              +--------------------------+
                       |
                       | TCP/IP received
                       |
              +--------------------------+
              |    Remote Endpoint       |
              |  (TCP/IP to Multicast)   |
              +--------------------------+
                       |
                       | multicast packets
                       |
              +--------------------------+
              |   Target Multicast       |
              |    Receiver(s)           |
              +--------------------------+
```

Not sure how you would go about translating this back to multicast on the target ios/android device tho. Might need to develop the transaltion yourself or find an app that already does this. Might even require a rooted device.

**Reply by eskimos20** [reply]
Thanks for the info! 

I think it is a bit to complicated for me, but it might work so if someone tries it It would be nice to know it works! :)
